In [32]:
import pandas as pd
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
import numpy as np
import joblib
from sqlalchemy import create_engine

In [ ]:
# Note:: The make sure you use the information from your specific PostgreSQL installation
host = r'127.0.0.1' # denotes that the db in a local installation
db = r'MSDS610' # db we just created
user = r'postgres' # using the postgres user for this demo
pw = r'apurba07' # this is the password established during installation
port = r'5432' # default port estabalished during install

In [34]:
db_conn = create_engine("postgresql://{}:{}@{}:{}/{}".format(user, pw, host, port, db))

In [35]:
table_name1 = r'feature_cols'
table_name2 = r'target_cols'
schema = r'analytics' # schema were the data was loaded last week.

X_val = pd.read_sql_table(table_name1, db_conn, schema)
y_val = pd.read_sql_table(table_name2, db_conn, schema)

In [36]:
table_name1 = r'features_cols'
table_name2 = r'targets_cols'
schema = r'analytics' # schema were the data was loaded last week.

X_val = pd.read_sql_table(table_name1, db_conn, schema)
y_val = pd.read_sql_table(table_name2, db_conn, schema)

In [37]:
X_val.head()

Age  Gender  Gaming_App_Usage_Hours  Number_of_Apps_Used  \
0  20.0       1                    4.66                 13.0   
1  53.0       0                    1.16                 20.0   
2  55.0       1                    1.83                 14.0   
3  24.0       0                    2.04                 25.0   
4  40.0       0                    2.06                 24.0   

   Total_App_Usage_Hours  
0                   3.95  
1                   5.92  
2                   1.82  
3                   4.97  
4                  10.70

In [38]:
y_val.head()

Productivity_App_Usage_Hours
0                          4.80
1                          4.84
2                          4.43
3                          1.05
4                          3.78

In [39]:
import joblib

best_model = joblib.load("best_model.pkl")



Here, we loaded the saved model that achieved the highest accuracy in the previous file.

In [40]:
# Generate predictions
y_val_pred = best_model.predict(X_val)

In [41]:
# Calculate MSE
mse = mean_squared_error(y_val, y_val_pred)
r2 = r2_score(y_val, y_val_pred)


In [42]:
print(f" Mean Squared Error (MSE): {mse}")
print(f" R² Score: {r2}")

 Mean Squared Error (MSE): 2.2537280141935487
 R² Score: 0.02421899049230314


We loaded the model and validated it in this file. In the previous file, we evaluated two different features with three different models and selected the best-performing model from our first feature list, as it provided the best validation results.

We used the same model here and analyzed its performance. The Mean Squared Error (MSE) of 2.2537 indicates that, on average, the squared differences between the predicted and actual values are relatively high, suggesting a moderate level of prediction error. The R² Score of 0.0242 suggests that the model explains only about 2.42% of the variance in the data, indicating a weak fit. 